src/aleph/web/controllers/accounts.py

```diff
@@ -62,19 +64,24 @@ def _get_address_from_request(request: web.Request) -> str:
 async def get_account_balance(request: web.Request):
     address = _get_address_from_request(request)
 
+    try:
+        query_params = GetAccountQueryParams.parse_obj(request.query)
+    except ValidationError as e:
+        raise web.HTTPUnprocessableEntity(text=e.json(indent=4))
+
     session_factory: DbSessionFactory = get_session_factory_from_request(request)
     with session_factory() as session:
-        balance = get_total_balance(
-            session=session, address=address, include_dapps=False
-        )
+        balance_detail = get_total_detailed_balance(session=session, address=address)
+        if query_params.chain is None:
+            balance = Decimal(sum(balance_detail.values()))
+            details = balance_detail
+        else:
+            balance = balance_detail.get(query_params.chain, Decimal(0))
+            details = {}
```

Comment:
That's not 100% super important but we probably want to do all those operations directly in SQL instead of doing them in python for optimization and clean code reasons.